lib/OpenQA/Shared/Controller/Auth.pm

```diff
@@ -56,7 +56,10 @@ sub auth ($self) {
     else {
 
         # Personal access token
-        if (my $userinfo = $self->req->url->to_abs->userinfo) {
+        if (($self->req->headers->authorization // '') =~ /^Bearer\s+(.+)$/) {
+            ($user, $reason) = $self->_token_auth($reason, $1);
```

Comment:
One question: if authentication is achieved here, shouldnt stop `_token_auth` repeating the steps for different token?
It seems that there is a possibility to run twice the ` _token_auth` with the `elsif`

Reply:
not sure I can but what do you think about (not sure if this actually works) but it should make sense

```
my $userinfo = (($self->req->headers->authorization) =~ /^Bearer\s+(.+)$/) ? $1 : $self->req->url->to_abs->userinfo;
($user, $reason) = $self->_token_auth($reason, $userinfo);
```

but not sure if it is more readable, to behonest